src/StudioCore/Utilities/ParamValidationTest.cs

```diff
@@ -0,0 +1,226 @@
+                PlatformUtils.Instance.MessageBox($"Param Load failed: {param}: {e.Message}", "Warning", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+            }
+        }
+        // ER
+        if (Locator.ActiveProject.Type is GameType.EldenRing)
+        {
+            try
+            {
+                using BND4 bnd = SFUtil.DecryptERRegulation(param);
+                LoadParamFromBinder(bnd, ref _params, out _paramVersion, true);
+            }
+            catch (Exception e)
+            {
+                PlatformUtils.Instance.MessageBox($"Param Load failed: {param}: {e.Message}", "Warning", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+            }
+        }
+        // AC6
+        if (Locator.ActiveProject.Type is GameType.ArmoredCoreVI)
+        {
+            try
+            {
+                using BND4 bnd = SFUtil.DecryptAC6Regulation(param);
+                LoadParamFromBinder(bnd, ref _params, out _paramVersion, true);
+            }
+            catch (Exception e)
+            {
+                PlatformUtils.Instance.MessageBox($"Param Load failed: {param}: {e.Message}", "Warning", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+            }
+        }
+    }
+
+    private static void LoadParamFromBinder(IBinder parambnd, ref Dictionary<string, Param> paramBank, out ulong version,
+        bool checkVersion = false, bool validatePadding = false)
+    {
+        var success = ulong.TryParse(parambnd.Version, out version);
+        if (checkVersion && !success)
+        {
+            throw new Exception(@"Failed to get regulation version. Params might be corrupt.");
+        }
+
+        // Load every param in the regulation
+        foreach (BinderFile f in parambnd.Files)
+        {
+            var paramName = Path.GetFileNameWithoutExtension(f.Name);
+
+            if (!f.Name.ToUpper().EndsWith(".PARAM"))
+            {
+                continue;
+            }
+
+            PARAM p;
+
+            p = PARAM.ReadIgnoreCompression(f.Bytes);
+            if (!_paramdefs.ContainsKey(p.ParamType ?? ""))
+            {
+                TaskLogs.AddLog(
+                    $"Couldn't find ParamDef for param {paramName} with ParamType \"{p.ParamType}\".",
+                    LogLevel.Warning);
+                continue;
+            }
+
+            if (p.ParamType == null)
+            {
+                throw new Exception("Param type is unexpectedly null");
+            }
+
+            PARAMDEF def = _paramdefs[p.ParamType];
+            try
+            {
+                p.ApplyParamdef(def);
+            }
+            catch (Exception e)
+            {
+                var name = f.Name.Split("\\").Last();
+                var message = $"Could not apply ParamDef for {name}";
+
+                TaskLogs.AddLog(message,
+                        LogLevel.Warning, TaskLogs.LogPriority.Normal, e);
+            }
+
+            TaskLogs.AddLog($"{paramName} validated");
+        }
+    }
+}
```

Comment:
Again, this is a solid utility to actually add to the program instead of having half-bakes across everyone's own workspaces.
Future work could include checking all our pads/dummy8s to see if they really are pad/dummy8
Nonetheless, a massive help